nvm.sh

```diff
@@ -2419,13 +2421,53 @@ nvm() {
             PROVIDED_REINSTALL_PACKAGES_FROM="$(nvm_echo "$1" | command cut -c 22-)"
             REINSTALL_PACKAGES_FROM="$(nvm_version "$PROVIDED_REINSTALL_PACKAGES_FROM")" ||:
           ;;
+          --skip-default-packages)
+            SKIP_DEFAULT_PACKAGES=true
+          ;;
           *)
             ADDITIONAL_PARAMETERS="$ADDITIONAL_PARAMETERS $1"
           ;;
         esac
         shift
       done
 
+      if [ -z "$SKIP_DEFAULT_PACKAGES" ] && [ -f "${NVM_DIR}/default-packages" ]; then
+        local DEFAULT_PACKAGES
+        DEFAULT_PACKAGES=""
+
+        # Read lines from $NVM_DIR/default-packages
+        local line
+        while IFS=" " read -r line; do
+
+          # Skip empty lines.
+          [ -z "${line}" ] || continue
+
+          # Skip comment lines that begin with `#`.
+
+          [ "$($line | head -c 1)" != "#" ] || continue
+
+          # Fail on lines that have multiple space-separated words
+          case ${VAR} in
+            *\ * )
+              nvm_err "Only one package per line is allowed in the ${NVM_DIR}/default-packages file. Please remove any lines with multiple space-seperated values."
+              return 1
+            ;;
+          esac
+
+          DEFAULT_PACKAGES="${DEFAULT_PACKAGES}${line} "
+        done < "${NVM_DIR}/default-packages"
+
+        if [ -z "$DEFAULT_PACKAGES" ]; then
+          nvm_echo "Installing default global packages from ${NVM_DIR}/default-packages..."
+          { 
+            nvm_echo "$DEFAULT_PACKAGES" | command xargs npm install -g --quiet 
```

Comment:
ah - you're trying to `npm install` before `nvm use` is ran on line 2488.

I think you should extract this block of code to a separate function; and then call that function on line 2495.5.